.github/workflows/deploy.yml

```diff
@@ -1,36 +1,123 @@
+                 echo "TF_VAR_SHARED_SECRET=${{              secrets.SHARED_SECRET       }}" >> $GITHUB_ENV
+                 echo "TF_VAR_NOTIFY_API_KEY=${{             secrets.NOTIFY_API_KEY      }}" >> $GITHUB_ENV
+                 echo "TF_VAR_GOOGLE_API_KEY=${{             secrets.GOOGLE_API_KEY      }}" >> $GITHUB_ENV
+                 echo "TF_VAR_TOTP_SECRET_KEY=${{            secrets.TOTP_SECRET_KEY     }}" >> $GITHUB_ENV
+                 echo "TF_VAR_SENTRY_DSN=${{                 secrets.SENTRY_DSN          }}" >> $GITHUB_ENV
+                 echo "TF_VAR_paas_logging_endpoint_port=${{ secrets.LOGIT_ENDPOINT      }}" >> $GITHUB_ENV
+             fi
+
+             if [ "${{github.event.inputs.environment }}" == "Production" ]
+             then
+                 echo ::set-output name=ARM_ACCESS_KEY::$(echo "${{ secrets.PROD_ARM_ACCESS_KEY  }}" )
+                 echo ::set-output name=control::$(echo "production" )
+                 echo ::set-output name=healthcheck::$(echo "${{env.PAAS_APPLICATION_NAME}}-prod" )
+                 echo ::set-output name=key::"api.production.terraform"
+                 echo ::set-output name=docker_image::"${{env.DOCKERHUB_REPOSITORY}}:sha-${{ steps.sha.outputs.short}}"
+
+                 echo "TF_VAR_CRM_CLIENT_ID=${{              secrets.PROD_CRM_CLIENT_ID       }}" >> $GITHUB_ENV
+                 echo "TF_VAR_CRM_CLIENT_SECRET=${{          secrets.PROD_CRM_CLIENT_SECRET   }}" >> $GITHUB_ENV
+                 echo "TF_VAR_CRM_TENANT_ID=${{              secrets.PROD_CRM_TENANT_ID       }}" >> $GITHUB_ENV
+                 echo "TF_VAR_CRM_SERVICE_URL=${{            secrets.PROD_CRM_SERVICE_URL     }}" >> $GITHUB_ENV
+                 echo "TF_VAR_GOOGLE_API_KEY=${{             secrets.PROD_GOOGLE_API_KEY      }}" >> $GITHUB_ENV
+                 echo "TF_VAR_SHARED_SECRET=${{              secrets.PROD_SHARED_SECRET       }}" >> $GITHUB_ENV
+                 echo "TF_VAR_NOTIFY_API_KEY=${{             secrets.PROD_NOTIFY_API_KEY      }}" >> $GITHUB_ENV
+                 echo "TF_VAR_TOTP_SECRET_KEY=${{            secrets.PROD_TOTP_SECRET_KEY     }}" >> $GITHUB_ENV
+                 echo "TF_VAR_SENTRY_DSN=${{                 secrets.SENTRY_DSN          }}" >> $GITHUB_ENV
+                 echo "TF_VAR_paas_logging_endpoint_port=${{ secrets.LOGIT_ENDPOINT      }}" >> $GITHUB_ENV
+             fi
 
-       - uses: softprops/turnstyle@v1
-         name: Wait for other inprogress deployment runs
+
+       - uses: hashicorp/setup-terraform@v1.2.1
          with:
-            continue-after-seconds: 180
+              terraform_version: 0.13.4
+
+       - name: Wait for any previous runs to complete
+         uses: softprops/turnstyle@v1
          env:
-           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+            GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+         with:
+           continue-after-seconds: 180
```

Comment:
Isn't it dangerous? We should fail after a long timeout, like 10-15min. And never continue.

Reply:
actually that one can be removed there is a turnstyle at the begining